src/profile.js

```diff
@@ -79,6 +86,39 @@ import {withRouter} from 'react-router-dom'
       });
     }
 
+    checkAge = () => {
```

Comment:
I think you should implement another function that checks if the inputed birthday is a valid date (refer to https://stackoverflow.com/questions/10589732/checking-if-a-date-is-valid-in-javascript possibly), instead of trying to implement checkAge and check valid date in the same function